component/componentstatus/status.go

```diff
@@ -0,0 +1,128 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package componentstatus // import "go.opentelemetry.io/collector/component/componentstatus"
+
+import (
+	"time"
+)
+
+// Watcher is an extra interface for Extension hosted by the OpenTelemetry
+// Collector that is to be implemented by extensions interested in changes to component
+// status.
+type Watcher interface {
```

Comment:
I would move this + the `InstanceId` to the "extension/statuswatcher` or something like that to be consistent with storage and other type of extensions.

Reply:
I think you're probably correct, but I'd prefer to encompass all public component status reporting features behind `componentstatus` for now while it is experimental and subject to lots of breaking API discussions.  Once we feel better about its API we could move it into `extension/storage` as a final location. Likely other aspects of `componentstatus` will move around in the future as well.

Reply:
Not suggesting to `extension/storage` but to `extension/statuswatcher` :)

Reply:
I am confused about why we do this then. Is this to just remove unstable code (that is ok)? But still you should at least document all these things as todos (file issues) and then we can do it in a separate PR.

Reply:
Removing unstable code from `component` and `extension` is the goal.  Issue created: https://github.com/open-telemetry/opentelemetry-collector/issues/10764